subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/XCTestIntegrationTest.groovy

```diff
@@ -221,6 +222,6 @@ dependencies {
         succeeds("assemble")
 
         then:
-        notExecuted(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        skipped(":assemble")
```

Comment:
this doesn't check that other tasks didn't run. It checks only that assemble was present and all of its dependencies were up-to-date. We should use `result.assertTasksExecuted()` or similar for these kinds of checks.